src/ULog.ts

```diff
@@ -214,30 +215,28 @@ export class ULog {
   }
 
   timeRange(): Readonly<[bigint, bigint]> | undefined {
-    if (!this._timeIndex || this._timeIndex.length === 0) {
-      return undefined;
-    }
-    const start = this._timeIndex[0]![0];
-    const end = this._timeIndex[this._timeIndex.length - 1]![0];
-    return [start, end];
+    return this._dataTimeRange;
   }
 
   private async createIndex(): Promise<void> {
     const timeIndex: IndexEntry[] = [];
     const dataCounts = new Map<number, number>();
+    let minTimestamp: bigint | undefined;
     let maxTimestamp = 0n;
     let logMessageCount = 0;
 
     let i = 0;
     let message: DataSectionMessage | undefined;
     while ((message = await this.readParsedMessage())) {
       if (message.type === MessageType.Data) {
+        minTimestamp ??= message.value.timestamp;
```

Comment:
If we are never checking with `<`, how do we know this is the min?

Reply:
arg, good catch! I forgot that these messages are not necessarily ordered